radix-engine/src/blueprints/epoch_manager/validator.rs

```diff
@@ -456,6 +457,168 @@ impl ValidatorBlueprint {
         Ok(())
     }
 
+    /// Locks the given stake units in an internal "delayed withdrawal" vault (which is the owner's
+    /// way of showing their commitment to running this validator in an orderly fashion - see
+    /// [`ValidatorSubstate.locked_owner_stake_unit_vault_id`]).
+    pub fn lock_owner_stake_units<Y>(
+        stake_unit_bucket: Bucket,
+        api: &mut Y,
+    ) -> Result<(), RuntimeError>
+    where
+        Y: ClientApi<RuntimeError>,
+    {
+        let handle = api.actor_lock_field(
+            OBJECT_HANDLE_SELF,
+            ValidatorField::Validator.into(),
+            LockFlags::read_only(),
+        )?;
+        let substate: ValidatorSubstate = api.field_lock_read_typed(handle)?;
+
+        Vault(substate.locked_owner_stake_unit_vault_id).put(stake_unit_bucket, api)?;
+
+        api.field_lock_release(handle)?;
+        Ok(())
+    }
+
+    /// Starts the process of unlocking the owner's stake units stored in the internal vault.
+    /// The requested amount of stake units (if available) will be ready for withdrawal after the
+    /// network-configured [`EpochManagerConfigSubstate.num_owner_stake_units_unlock_epochs`] via a
+    /// call to [`finish_unlock_owner_stake_units()`].
+    /// As an optimization, this method also performs the same "finish unlocking" operation, and
```

Comment:
I don't quite like the optimisation here. 
- It makes the interface less intuitive and force users to handle the returned bucket
- It may even downgrade the performance due to extra `take` invocations, which can batched in `finish_unlock`.

Reply:
Your arguments are very valid.

I remember that this "optimization" idea came around somewhere on Slack (probably from David/Matt?) in order to mitigate the problem of "max pending withdrawals limit" (i.e. owner should be able to do "continuous withdrawal" easily).

One improvement that I see here would be to NOT return the alread-unlocked bucket, but instead move the amount from the `pending_owner_stake_unit_withdrawals` queue to a new, simple field (like, `already_unlocked_owner_stake_unit_amount: Decimal`). And on `finish_unlock`, the logic would simply take this field into account too.

This should actually address both of your points (easier API + no `.take()` in the `start_unlock`) for a little price of slightly-more-complex `finish_unlock` :) let me try...

Reply:
I have applied the idea from https://github.com/radixdlt/radixdlt-scrypto/pull/1037#discussion_r1201909487 - now the API is cleaner (no auto-finish-unlock-as-a-drive-by anymore). Please have a look @iamyulong 

I am still waiting for opinions about "let's not have any limit at all" - if more people vote that we can do it, I will do it (and then ofc this code will be ditched).

Reply:
Alright, so I realized that this collection will never reach the 4MB substate limit (in fact, it won't grow larger than a number of owner stake withdrawal epochs delay - because we stack withdrawals scheduled for the same epoch! see `multiple_pending_owner_stake_unit_withdrawals_stack_up()`).

This means we can remove the explicit limit too, without much risk of a performance hit. Done 👍